src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -621,4 +623,72 @@ BOOST_AUTO_TEST_CASE(OptimizerTest)
   model.Train(trainData, trainLabels, opt);
 }
 
+/**
+ * Train the RBF network on a larger dataset.
+ */
+BOOST_AUTO_TEST_CASE(RBFNetworkTest)
+{
+  // Load the dataset.
+  arma::mat trainData;
+  data::Load("thyroid_train.csv", trainData, true);
+
+  arma::mat trainLabels = trainData.row(trainData.n_rows - 1);
+  trainData.shed_row(trainData.n_rows - 1);
+
+  arma::mat testData;
+  data::Load("thyroid_test.csv", testData, true);
+
+  arma::mat testLabels = testData.row(testData.n_rows - 1);
+  testData.shed_row(testData.n_rows - 1);
+
+  /*
+   * Construct a feed forward network with trainData.n_rows input nodes,
+   * hiddenLayerSize hidden nodes and trainLabels.n_rows output nodes. The
+   * network structure looks like:
+   *
+   *  Input         RBF          Activation    Output
+   *  Layer         Layer         Layer        Layer
+   * +-----+       +-----+       +-----+       +-----+
+   * |     |       |     |       |     |       |     |
+   * |     +------>|     +------>|     +------>|     |
+   * |     |       |     |       |     |       |     |
+   * +-----+       +--+--+       +-----+       +-----+
+   */
+
+  arma::mat centroids;
+  KMeans<> kmeans;
+  kmeans.Cluster(trainData, 8, centroids);
+
+  FFN<NegativeLogLikelihood<> > model;
+  model.Add<RBF<> >(trainData.n_rows, 8, centroids);
+  model.Add<Linear<> >(8, 3);
+  model.Add<LogSoftMax<> >();
+
+  TestNetwork<>(model, trainData, trainLabels, testData, testLabels, 10, 0.1);
+  arma::mat dataset;
+  dataset.load("mnist_first250_training_4s_and_9s.arm");
+
+  // Normalize each point since these are images.
+  for (size_t i = 0; i < dataset.n_cols; ++i)
+  {
+    dataset.col(i) /= norm(dataset.col(i), 2);
+  }
+
+  arma::mat labels = arma::zeros(1, dataset.n_cols);
+  labels.submat(0, labels.n_cols / 2, 0, labels.n_cols - 1).fill(1);
+  labels += 1;
+
+  arma::mat centroids1;
+  arma::Row<size_t> assignments;
+  KMeans<> kmeans1;
+  kmeans1.Cluster(dataset, 10, centroids1);
+
+  FFN<NegativeLogLikelihood<> > model1;
+  model1.Add<RBF<> >(dataset.n_rows, 10, centroids1);
+  model1.Add<Linear<> >(10, 2);
+  model1.Add<LogSoftMax<> >();
```

Comment:
I don't think we need the LogSoftMax layer if we will use the `MeanSquaredError`